features/dd-sdk-android-trace-otel/src/test/kotlin/com/datadog/android/trace/opentelemetry/internal/MiscUtilsTest.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.opentelemetry.internal
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.trace.opentelemetry.utils.verifyLog
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.tools.unit.annotations.TestTargetApi
+import com.datadog.tools.unit.extensions.ApiLevelExtension
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(ApiLevelExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class MiscUtilsTest {
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Test
+    fun `M return defaultValue and log exception W executeSafelyOnAndroid23AndBelow{action throws exception, below 24}`(
+        @StringForgery defaultValue: String
+    ) {
+        // Given
+        val exception = Exception()
+        val action: () -> String = { throw exception }
```

Comment:
Let's have a chat about this as I am not sure I understand exactly what you mean. One note from my part is that we will not be able to check the whole package existence through reflection unless we will go one by one. The more we discuss about this the more I think that maybe this safety mechanism is not the best idea and we should just go with desugaring required.

Reply:
Let's say Otel code is updated with some class which exists on say API 25. `Class.forName("function")` will be fine, but when `action()` is executed it will be `NoClassDefFound**Error**` for this new class, not exception. But imo with current state of the code it is fine.